tests/base/function_parser_09.cc

```diff
@@ -0,0 +1,58 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2005 - 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// test rand function
+
+#include "../tests.h"
+#include <fstream>
+#include <iomanip>
+#include <map>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/point.h>
+#include <deal.II/lac/vector.h>
+#include <deal.II/base/function_parser.h>
+
+
+void eval(const std::string &exp)
+{
+  std::string variables = "x,y";
+  std::map<std::string,double> constants;
+
+  FunctionParser<2> fp(1);
+  fp.initialize(variables,
+                exp,
+                constants);
+
+  Point<2> p;
+
+  deallog << fp.value(p) << std::endl;
+}
+
+
+void test()
+{
+  //  eval("rand()"); //random seed
+  eval("rand_seed(10)");
```

Comment:
Could you also add a test that evaluates the `rand()` function just to make sure that the call succeeds? It will have to discard the value since it will not be reproducible, but it could test for its value being between 0 and 1.